datalake/src/main/java/com/sequenceiq/datalake/service/validation/cloudstorage/CloudStorageValidator.java

```diff
@@ -122,6 +145,22 @@ private ObjectStorageValidateRequest createObjectStorageValidateRequest(
         return result.build();
     }
 
+    private ObjectStorageValidateRequest createBackupLocationValidateRequest(
+            CloudCredential credential, CloudStorageRequest cloudStorageRequest,
+            DetailedEnvironmentResponse environment) {
+        cloudStorageRequest.setLocations(Collections.emptyList());
```

Comment:
This side-effect is unexpected. I understand that the `CloudStorageRequest` instance gets retrieved from the JSON-deserialized `StackV4Request`, but it is still that might be worth flagging directly for sake of warning the caller. Could you please add a note for this in the javadoc of `validateBackupLocation()` above?

Note: Ideally this input arg should not be modified at all, rather be "cloned" (either via a proper `clone()` or some alternative approach) and then the cloned copy adjusted as needed. On the other hand, I am fine with the current approach as long as there is a visible warning for the side-effect like mentioned above.

Reply:
Can you help me understand why cloudStorageRequest should not be modified?

Reply:
It is usually not a good practice to make changes to the object contents of any input args (of a mutable reference type), unless the intention can be inferred from the method naming or its javadoc. In this case, this is definitely not a big deal because the `CloudStorageRequest` is a throw-away instance anyway (created afresh from the JSON). Here, I would not expect such a change by just looking at the method name.

Reply:
Thanks for the clarification.

Reply:
Note: Instead of a javadoc suggested above, an alternative would be changing the code (in `validateBackupLocation()`) so that it saves the original `CloudStorageRequest.locations` before the `createBackupLocationValidateRequest()` invocation and restores it afterwards. That way your tweak would be temporary and totally transparent to the outside caller. :smile:

Reply:
done